lib/Parser/Parse.cpp

```diff
@@ -1187,40 +1128,43 @@ ParseNodePtr Parser::CreateBlockNode(charcount_t ichMin,charcount_t ichLim, Pnod
 ParseNodePtr Parser::CreateStrNode(IdentPtr pid)
 {
     Assert(!this->m_deferringAST);
-
-    ParseNodePtr pnode = CreateNode(knopStr);
-    pnode->AsParseNodePid()->pid=pid;
+    ParseNodePtr pnode = Anew(&m_nodeAllocator, ParseNodePid, knopStr, this->GetScanner()->IchMinTok(), this->GetScanner()->IchLimTok(), pid);
     pnode->grfpn |= PNodeFlags::fpnCanFlattenConcatExpr;
+    AddAstSize(sizeof(ParseNodePid));
     return pnode;
 }
 
 ParseNodePtr Parser::CreateIntNode(int32 lw)
 {
-    ParseNodePtr pnode = CreateNode(knopInt);
-    pnode->AsParseNodeInt()->lw = lw;
+    Assert(!this->m_deferringAST);
+    ParseNodeInt * pnode = Anew(&m_nodeAllocator, ParseNodeInt, this->GetScanner()->IchMinTok(), this->GetScanner()->IchLimTok(), lw);
+    AddAstSize(sizeof(ParseNodeInt));
     return pnode;
 }
 
-// Create Node with scanner limit
-template <OpCode nop>
-ParseNodePtr Parser::CreateNodeWithScanner()
-{    
-    return CreateNodeWithScanner<nop>(this->GetScanner()->IchMinTok());
+ParseNodePtr Parser::CreateNameNode(IdentPtr pid)
```

Comment:
These Create* methods could return a pointer to the type they Anew'd (ParseNodePid* here) instead of generic ParseNodePtr. Would save some callers from having to reinterpret_cast the return values. Could also move these into their respective classes as static helpers as well. I'm fine with this code as is - it's a big improvement - but there is still a lot more we can do around here in future. 😁

Reply:
You read my mind.  That will be in the future.  Just don't want to generate too much diff to make it hard to review.